### PR TITLE
EFRS-1159-BugFix  Prevent non valid identical files to be saved in th…

### PR DESCRIPTION
**Pull request by smchedlidze826**

### CODE REVIEW

File: ui/src/app/store/manage-collectiom/reducers.ts [hunk 1]
```diff
@@ -160,10 +160,19 @@ const reducer: ActionReducer<CollectionEntityState> = createReducer(
...
 
   on(getSubjectExamplesFail, state => ({ ...state, isCollectionPending: false })),
-  on(addFileToCollection, (state, { url, file, subject }) => ({
-    ...state,
-    collection: [...state.collection, { url, file, subject, status: CircleLoadingProgressEnum.OnHold }],
-  })),
+
+  on(addFileToCollection, (state, { url, file, subject }) => {
+    const collectionCopy = [
+      ...state.collection.filter(item => item.status !== CircleLoadingProgressEnum.Failed && item.subject === state.subject),
```
Comment (isobolev) [inline ui/src/app/store/manage-collectiom/reducers.ts:166]: filter returns new array, no need to use [...state.collection.filter] construction, can be just state.collection.filter

File: ui/src/app/store/manage-collectiom/reducers.ts [hunk 2]
```diff
@@ -160,10 +160,19 @@ const reducer: ActionReducer<CollectionEntityState> = createReducer(
...
+    ];
+
+    const failedCollection = [...state.collection.filter(item => item.status === CircleLoadingProgressEnum.Failed && item.url !== url)];
+
+    return {
+      ...state,
+      collection: [...collectionCopy, ...failedCollection, { url, file, subject, status: CircleLoadingProgressEnum.OnHold }],
```
Comment (isobolev) [inline ui/src/app/store/manage-collectiom/reducers.ts:173]: we can also avoid assigning memory to arrays which we get rid of right away. how about something like this?
```
return {
    ...state,
    collection: [
        ...state.collection.filter(item => 
            (item.status !== CircleLoadingProgressEnum.Failed && item.subject === state.subject) // not failed
            || (item => item.status === CircleLoadingProgressEnum.Failed && item.url !== url) // failed
        ), { url, file, subject, status: CircleLoadingProgressEnum.OnHold }
    ],
}
```

Comment (smchedlidze826) [reply ui/src/app/store/manage-collectiom/reducers.ts:173]: fixed

File: ui/src/app/store/manage-collectiom/reducers.ts (hunk 1 above)
Comment (smchedlidze826) [reply ui/src/app/store/manage-collectiom/reducers.ts:166]: fixed

File: ui/src/app/store/manage-collectiom/reducers.ts (hunk 2 above)
Comment (smchedlidze826) [reply ui/src/app/store/manage-collectiom/reducers.ts]: fixed

**Review by isobolev — APPROVED** [review]

**Pull request merged**